draft-trammell-taps-post-sockets.md

```diff
@@ -818,12 +830,118 @@ func sayHelloWithAssociation(association Association) {
+is not available on Sources. Carriers also provide .OnSent(), .OnAcked(), and
+.OnExpired() calls for binding default send event handlers to the Carrier, and
+.OnClosed() for handling passive close notifications.
+
+~~~~~~~
+                                  +---------[incoming]-----------+
+                                  |         [Message ]           V
+[outgoing] ---> .Send() --->  [Carrier] <---- .Ready() <---- [Receiver]
+[Message ]                        |
+                                  +--- .OnSent()
+                                  +--- .OnAcked()
+                                  +--- .OnExpired()
+                                  +--- .OnClosed()       
+~~~~~~~
+{: #fig-message-lc title="Sending and Receiving Messages and Events"}
+
+
+\[EDITOR'S NOTE (bht) on the text below: Parameters, as above, PolicyContext, and
+a Configuration are the same thing, as in
+https://github.com/mami-project/draft-trammell-post-sockets/issues/15 -- I'll
+use Configuration here, since that's where we're going, but won't modify this in
+the rest of the document]. 
+
+An application may have a global Configuation, as well as more specific
+Configurations to apply to the establishment of a given Association or Carrier.
+These Configurations are optional arguments to the Association and Carrier
+creation calls. Each Configuration is conceptually a key-value store, where
+information in more specific scopes overrides information in less specific
+scopes: application defaults can be overriden by specific Configurations bound
+to Carriers or Associations, all of which may be overriden by system or
+user-scoped configuration parameters. Configurations are also made directly
+available to protocol stack instances (PSIs, see {{sec-psi}}) for fine-grained
+control of implementation-specific configuration parameters.
+
+In order to initiate a connection with a remote endpoint, a user of Post Sockets
+must start from a Remote (see {{remote}}). A Remote encapsulates identifying
+information about a remote endpoint at a specific level of resolution. A new
+Remote can be wrapped around some identifying information by via the NewRemote()
+call. A Remote has a .Resolve() method, which can be iteratively revoked to
+increase the level of resolution; a call to Resolve on a given Remote may result
+in one to many Remotes, as shown in {{fig-remote-lc}}.  Remotes at any level of
+resolution may be passed to Post Sockets calls; each call will continue
+resolution to the point necessary to establish or resume a Carrier.
+
+~~~~~~~
+                          +----------------------------+
+                        n |                            | 1
+NewRemote(identifiers) ---+--->[Remote] --.Resolve()---+
+~~~~~~~
+{: #fig-remote-lc title="Recursive resolution of Remotes"}
+
+Information about the local endpoint is also necessary to establish
+communication. This is passed in the form of a Local (see {{local}}). A
```

Comment:
s/establish communication./establish an association which implicitly will happen if a new carrier is initiated and no association with a respective local-remote pair exists or the configuration explicitly required the establishment of a new association./

Reply:
will fix. (the second sentence is a bit unwieldy, but I get the point)